### PR TITLE
remove some InTreePluginXXXUnregister

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Instead of https://github.com/kubernetes/kubernetes/pull/124023, This PR deletes the feature gates and all their usage. It's small and we can argue if it's safe or not to remove them.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Removed the following feature gates:
- `InTreePluginAWSUnregister`
- `InTreePluginAzureDiskUnregister`
- `InTreePluginAzureFileUnregister`
- `InTreePluginGCEUnregister`
- `InTreePluginOpenStackUnregister`
- `InTreePluginvSphereUnregister`
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
Action required: Alpha feature gates InTreePluginAWSUnregister, InTreePluginAzureDiskUnregister, InTreePluginAzureFileUnregister, InTreePluginGCEUnregister, InTreePluginOpenStackUnregister and InTreePluginvSphereUnregister were removed. These feature gates had no visible effect on a cluster since Kubernetes 1.30, where the in-tree volume plugins were already removed.
```

### CODE REVIEW

**Comment by xing-yang** [comment]
I think https://github.com/kubernetes/kubernetes/pull/124023 is safer. Then the feature gates can be removed in 1.32.

**Comment by carlory** [comment]
@xing-yang I reopen https://github.com/kubernetes/kubernetes/pull/124023. That PR needs https://github.com/kubernetes/kubernetes/pull/124108 merged that fixes unit test failures.

**Comment by carlory** [comment]
>@carlory can you please split it into two PRs, one that updates all the unit tests (most of this PR) and a second one that deletes the feature gates and all their usage? That one should be small and we can argue if it's safe or not to remove them.

Let's keep this PR open and wait feedback from @jsafrane

File: pkg/volume/csimigration/plugin_manager.go [hunk 1]
```diff
@@ -60,18 +60,6 @@ func (pm PluginManager) IsMigrationCompleteForPlugin(pluginName string) bool {
...
-	case csilibplugins.GCEPDInTreePluginName:
-		return pm.featureGate.Enabled(features.InTreePluginGCEUnregister)
-	case csilibplugins.AzureFileInTreePluginName:
-		return pm.featureGate.Enabled(features.InTreePluginAzureFileUnregister)
-	case csilibplugins.AzureDiskInTreePluginName:
-		return pm.featureGate.Enabled(features.InTreePluginAzureDiskUnregister)
-	case csilibplugins.CinderInTreePluginName:
-		return pm.featureGate.Enabled(features.InTreePluginOpenStackUnregister)
-	case csilibplugins.VSphereInTreePluginName:
-		return pm.featureGate.Enabled(features.InTreePluginvSphereUnregister)
```
Comment (jsafrane) [inline pkg/volume/csimigration/plugin_manager.go:74]: The function should return `true` for these cases.

**Review by jsafrane — COMMENTED** [review]
Sorry for a very late review.

File: pkg/volume/csimigration/plugin_manager.go (hunk 1 above)
Comment (carlory) [reply pkg/volume/csimigration/plugin_manager.go:74]: There's another PR to promote these feature gates to GA and lock it by default. https://github.com/kubernetes/kubernetes/pull/124023.  https://github.com/kubernetes/kubernetes/pull/124108 is to fix this test failure in https://github.com/kubernetes/kubernetes/pull/124023. 

Should we promote or delete this feature-gate from the codebase? @jsafrane

Comment (carlory) [reply pkg/volume/csimigration/plugin_manager.go:74]: If the promotion is desired, this PR can be closed.

Comment (jsafrane) [reply pkg/volume/csimigration/plugin_manager.go:74]: I added it to today's meeting agenda. I would personally delete them.

Comment (jsafrane) [reply pkg/volume/csimigration/plugin_manager.go:74]: We decided to continue with this PR and delete the feature gates right away. Please fix this comment and we merge it tomorrow.

**Comment by jsafrane** [comment]
/lgtm
/approve
Thanks for your patience!

**Pull request merged**